### ISSUE TITLE
MS Visual Studio and undesired warning C4068: unknown pragma

### ISSUE THREAD

**Issue by Thomas-Haase** [docopt/docopt.cpp#64]
When compiling the code with MS Visual Studio 2015 (Version 14.0.25123.00 Update 2) I get several undesired warnings C4068: unknown pragma. 

```
xxx\docopt\docopt_util.h(22): warning C4068: unknown pragma
xxx\docopt\docopt_util.h(23): warning C4068: unknown pragma
xxx\docopt\docopt_private.h(325): warning C4068: unknown pragma
xxx\docopt\docopt_private.h(326): warning C4068: unknown pragma
xxx\docopt\docopt.cpp(57): warning C4068: unknown pragma
xxx\docopt\docopt.cpp(58): warning C4068: unknown pragma
```

I know, this warnings are uncritical, but usually when compiling I prefer to see 0 warnings 0 errors.
It seems this issue is caused by pragmas which make happy some Apple related compilers like ObjectiveC. 
Is it possible to surround these pragmas with something like this?

``` cpp
#ifndef _MSC_VER
#pragma mark -
#pragma mark General utility
#endif
```

**Comment by jaredgrubb** [comment]
I think the idiom is to use "#if 0" around the pragma-mark. Alternatively, we could just remove them.

**Comment by Thomas-Haase** [comment]
Yes, removing would be fine for me. 
But I don't know if anybody in Apple environments will miss this pragma mark in his [XCode menu.](http://cocoasamurai.blogspot.de/2006/09/tip-pragma-mark-organizing-your-source.html) Probably a positive ifdef is more appropriate to  maky happy all:
Some macro [candidates ](https://developer.apple.com/library/content/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html#//apple_ref/doc/uid/TP40002850-SW13) may be `__APPLE_CC__` or `__OBJC__` or `__APPLE__`, but unfortunately I'm not an expert for this compilers, so I'm not sure which one or which combination is the right one.
```cpp
#ifdef __APPLE_CC__
#pragma mark -
#pragma mark General utility
#endif
```

**Comment by Thomas-Haase** [comment]
Hi, are there any chances to see the this patch or simple-fix in a new release?

**Comment by jaredgrubb** [comment]
Thanks to @nikodemus for a patch on this. This has been merged (PR#90).
